File: 1.二维数组中的查找.py
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target, array):
        # write code here
        row = len(array)
        col = len(array[0])

        i, j = 0, col - 1

        while i < row and j >= 0:
            if target == array[i][j]:
                return True
            if target < array[i][j]:
                j -= 1
            else:
                i += 1

        return False

File: 1.二维数组中的查找.cc
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        int row = array.size();
        int col = array[0].size();

        int i = 0, j = col - 1;
        while (i < row && j >= 0)
        {
            if (array[i][j] == target)
                return true;
            if (array[i][j] > target)
            {
                j -= 1;
            }else{
                i += 1;
            }
        }
        return false;
    }
};